### PR TITLE
refactor: migrate node-utils from commonjs to es6 module

### PR DESCRIPTION
**Pull request by armano2**
benefits: tree-shaking and faster access to imports

i'm going to wait with changing converter after #120 will be finished
```ts
import * as nodeUtils from './node-utils';
```

### CODE REVIEW

**Comment by JamesHenry** [comment]
No rush at all, I'll just wait until you let me know this PR is ready. I was literally thinking about this change just this morning - another legacy of the codebase being originally a pure JS package

File: src/convert.ts [hunk 1]
```diff
@@ -907,14 +907,14 @@ export default function convert(config: ConvertConfig): ESTreeNode | null {
...
         node.name,
         ast,
         (token: any) => {
           if (!token || !token.kind) {
             return false;
           }
-          return nodeUtils.getTextForTokenKind(token.kind) === '(';
+          return getTextForTokenKind(token.kind) === '(';
         },
         ast
```
Comment (armano2) [inline src/convert.ts]: btw. this function is expensive it's doing a lot of loops its i think it's not a best way to do this xd

Comment (JamesHenry) [reply src/convert.ts]: Hmm interesting, I would have thought a function like that would benefit from some optimizations in the JS engine, but you might be right.

Would definitely be interesting to add some benchmarking now that things are settling down a bit.

I would not feel confident making large-scale changes without that in place (obviously not referring to something minor like this).

Comment (armano2) [reply src/convert.ts]: no i'm not talking about execution but what is done in `findFirstMatchingToken` 

```
  while (previousToken) {
    if (predicate(previousToken)) {
      return previousToken;
    }
    previousToken = findNextToken(previousToken, parent, ast);
  }
  return undefined;

...

export function findNextToken(
  previousToken: ts.Node,
  parent: ts.Node,
  ast: ts.SourceFile
): ts.Node | undefined {
  return find(parent);

  function find(n: ts.Node): ts.Node | undefined {
    if (ts.isToken(n) && n.pos === previousToken.end) {
      // this is token that starts at the end of previous token - return it
      return n;
    }
    return firstDefined(n.getChildren(ast), (child: ts.Node) => {
      const shouldDiveInChildNode =
        // previous token is enclosed somewhere in the child
        (child.pos <= previousToken.pos && child.end > previousToken.end) ||
        // previous token ends exactly at the beginning of child
        child.pos === previousToken.end;
      return shouldDiveInChildNode && nodeHasTokens(child, ast)
        ? find(child)
        : undefined;
    });
  }
}
```

Comment (armano2) [reply src/convert.ts]: there is loop in `findFirstMatchingToken` -> loop in findNextToken -> and recursion

**Pull request merged**